curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-asynchronous-programming-by-building-an-fcc-forum-leaderboard/64440a50d076c204051f2cc0.md

```diff
@@ -1,46 +1,56 @@
 ---
 id: 64440a50d076c204051f2cc0
-title: Step 34
+title: Step 25
 challengeType: 0
-dashedName: step-34
+dashedName: step-25
 ---
 
 # --description--
 
 You need a function to convert view counts to a more readable format. For example, if the view count is `1000`, it should display as `1k` and if the view count is `100,000` it should display as `100k`.
 
-Create an arrow function called `viewCount` with a parameter called `views`.
+Create a `viewCount` function with a `views` parameter, then return the `views`.
+
+Inside the `viewCount` function, create a `thousands` variable and set it to the value of rounding down `views` based on a thousand.
+
+After that, check if `views` is greater than or equal to `1000`. If it is, use template literals to return the `thousands` variable in the format `${thousands}k`.
 
 # --hints--
```

Comment:
As for all of these tests, we should only care that they created the correct function name and test the final result.
This will allow for multiple correct solutions.

````md
You should have a function called `viewCount`.

```js
assert.isFunction(viewCount);
```

Your `viewCount` function should have a parameter called `views`.

```js
assert.match(viewCount.toString(), /\(views\)/);
```

Your `viewCount` function should return the string `1k` when `views` is `1000`.

```js
assert.strictEqual(viewCount(1000), "1k");
```

Your `viewCount` function should return the string `100k` when `views` is `100000`.

```js
assert.strictEqual(viewCount(100000), "100k");
```

Your `viewCount` function should return the correct string when `views` is greater than or equal to `1000`.

```js
assert.strictEqual(viewCount(10000), "10k");
assert.strictEqual(viewCount(35000), "35k");
assert.strictEqual(viewCount(99999), "99k");
```

Your `viewCount` function should return `views` when `views` is `200`.

```js
assert.strictEqual(viewCount(200), 200);
```

Your `viewCount` function should return `views` when `views` is less than `1000`.

```js
assert.strictEqual(viewCount(999), 999);
assert.strictEqual(viewCount(500), 500);
assert.strictEqual(viewCount(1), 1);
```
````

Those tests should allow for correct solutions like these 

````md
```js
const viewCount = (views)=> views >= 1000 ? `${Math.floor(views / 1000)}k` : views

```

```js
function viewCount(views) {

  if (views >= 1000) {
    return `${Math.floor(views / 1000)}k`;
  } else {

    return views
  }

};

```

```js
const viewCount = (views) => {
  const thousands = Math.floor(views / 1000);

  if (views >= 1000) {
    return `${thousands}k`;
  }

  return views;
};
```

Reply:
I missed this earlier, but this second test should have the parens be optional
```
assert.match(viewCount.toString(), /\(views\)/);
```